packages/gatsby-link/src/index.js

```diff
@@ -16,13 +24,23 @@ export function withPrefix(path) {
 }
 
 export function withAssetPrefix(path) {
-  return [__PATH_PREFIX__].concat([path.replace(/^\//, ``)]).join(`/`)
+  return isRelativePath(path)
+    ? path
+    : [__PATH_PREFIX__].concat([path.replace(/^\//, ``)]).join(`/`)
```

Comment:
_sort of_ but there's browser support to consider for `startsWith`. It's a better API (that may not have existed or been implemented broadly at the time of this code change), but I'm not convinced it's worth making the change when it's functionally the same thing and more supported. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

Reply:
I'd assume we're polyfilling it already

Reply:
🤷 are we comfortable going forward with an assumption?

Reply:
The reason I said I assumed it would be polyfilled is because the docs say we automatically include them. The benefit is performance. Regexes are really expensive. We could in theory replace it with a substring comparison instead, but that's a lot less readable, and by the time we've used it a few times will be larger than the polyfill (and unlike the polyfill it won't be removed if people target new browsers).

Reply:
My other suggestions use `startsWith` and `endsWith` as well, fwiw.

If this is really a concern then yeah, don't do it. I too thought the polyfills are in place for this.